lib/Driver/WindowsToolChains.cpp

```diff
@@ -186,6 +189,22 @@ toolchains::Windows::constructInvocation(const DynamicLinkJobAction &job,
   context.Args.AddAllArgs(Arguments, options::OPT_linker_option_Group);
   context.Args.AddAllArgValues(Arguments, options::OPT_Xclang_linker);
 
+  switch (context.OI.LTOVariant) {
+  case OutputInfo::LTOKind::LLVMThin:
+  case OutputInfo::LTOKind::LLVMFull: {
+    if (Linker.empty())
+      Arguments.push_back("-fuse-ld=lld");
```

Comment:
Perhaps you should move this condition out of the `switch` and make it `if (Linker.empty() && context.OI.LTOVariant != OutputInfo::LTOKind::None)`. Then you could get rid of the other `if` statement below and simply make the appropriate `Arguments.push_back()` call in each case, making the control flow much more straightforward.

For that matter, maybe you should move this all the way up to line 79 or so, where you can set `Linker` right before a `-fuse-ld=` would normally be generated, rather than adding another place it could be inserted down here and having to reason about whether or not you already did it earlier. Again, please either make `OPT_use_ld` "win" over this logic, or diagnose an error if you use both.